src/routers/semester.ts

```diff
@@ -26,43 +32,52 @@ semesterRouter.post('/', async (req, res) => {
 	try {
 		const userID = req.header('userID');
 		if (userID) {
-			const semester = new Semester({
-				owner: userID,
-				name: req.body.name,
-				courseItems: [],
-			});
-			const created = await create(semester);
-			res.json(created);
+			const body = req.body;
+			const checkPost = ajv.compile(
+				schema.components.schemas.SemesterCreate
+			);
+
+			if (checkPost(body)) throw Error('body invalid');
+
+			const semester = new Semester({ owner: userID, ...body });
+
+			console.log(semester); // TODO: call db operation
+			res.status(201); // + .json(created);
 		}
 	} catch (err) {
 		res.status(404).send(ERROR_RESPONSE);
 	}
 });
 
-semesterRouter.put('/', async (req, res) => {
+semesterRouter.put('/:id', async (req, res) => {
 	try {
+		const id = req.params.id;
 		const userID = req.header('userID');
-		if (userID) {
+		if (id && userID) {
 			const body = req.body;
-			const semester = new Semester({
-				id: body.id,
-				owner: userID,
-				name: body.name,
-			});
-			const updated = await update(semester, userID);
-			res.json(updated);
+			const checkPut = ajv.compile(
+				schema.components.schemas.SemesterUpdate
+			);
+
+			if (checkPut(body)) throw Error('body invalid');
+
+			const semester = new Semester(body);
+
+			console.log(semester); // TODO: call db operation
+			res.status(200); // + .json(updated)
 		}
```

Comment:
if `id` or `userId` don't exist, the request won't receive a response (we must call res.send() on every code path). Might be better to do `if(!id || !userID) throw Error("message")`

Reply:
good catch!